### ISSUE TITLE
<g:include tag stopped working in version 4.0.4

### ISSUE THREAD

**Issue by miguelabautista** [grails/grails-core#11601]
include tag is not working when running the app in a WAR. 

the tag works when running in development mode.

It's not working only in this version. I tested in version 4.0.3 and no problem.

### Task List

- [ ] Steps to reproduce provided

I created a sample app. create the war and run it using tomcat. 
     
- [ ] Stacktrace (if present) provided

WARN org.springframework.web.servlet.PageNotFound - No mapping for GET /prueb/prueb/dale/sect

- [ ] Example that reproduces the problem uploaded to Github

 https://github.com/miguelabautista/tagIssue

**Comment by lkuchars** [comment]
I'm having the same issue after upgrading from 4.0.3 to 4.0.4. Did you find your way around this?

**Comment by lkuchars** [comment]
@puneetbehl It seems to be related to this commit [https://github.com/grails/grails-gsp/commit/3f0d9bd4789f6d14910d419eef56e8e53ed33cf8#diff-63bca790b2fb44d7ef2f3c9b99d977fc](https://github.com/grails/grails-gsp/commit/3f0d9bd4789f6d14910d419eef56e8e53ed33cf8#diff-63bca790b2fb44d7ef2f3c9b99d977fc)

I stepped it in debugger with 4.0.3 vs 4.0.4. Newer uses version after the change above which in turn uses link generator. In my case it's CachingLinkGenerator. Returned include link contains context path. In my case ```/backoffice/``` In previous version of grails url did not have context. Later at internal forward tomcat adds another ```/backoffice/``` to this and URL is not found in effect

![Screenshot_20200903_123520](https://user-images.githubusercontent.com/24829139/92105094-9c5db480-ede2-11ea-945e-b5847f893dce.png)
![Screenshot_20200903_123617](https://user-images.githubusercontent.com/24829139/92105100-9ff13b80-ede2-11ea-9575-5757d58a8916.png)

**Comment by gelleouet** [comment]
1/ When include tag is used with controller and action attributes, `org.grails.web.mapping.UrlMappingUtils#buildDispatchUrlForMapping` create a Map with these 2 attributes and delegate it to `linkGenerator`.

2/ include context is activated by default in `org.grails.web.mapping.DefaultLinkGenerator#link` method and url is built with context and so is absolute :
`boolean includeContext = GrailsClassUtils.getBooleanFromMap(ATTRIBUTE_INCLUDE_CONTEXT, attrs, true)`

3/ `org.grails.web.mapping.UrlMappingUtils#includeForUrl` method get a `RequestDispatcher` on an absolute include url but `javax.servlet.ServletRequest#getRequestDispatcher` says :

> The pathname specified may be relative, although it cannot extend outside the current servlet context.  If the path begins with a "/" it is interpreted as relative to the current context root.

So the dispatcher append its context (the same) to the absolute url. 

A simple fix could be to create a Map in `org.grails.web.mapping.UrlMappingUtils#buildDispatchUrlForMapping` with an addittionnal parameter `includeContext=false` to force `linkGenerator` to create an relative url and to be compatible with request dispatcher ?

**Comment by gelleouet** [comment]
If you change `server.servlet.context-path` properties, don't forget to change also `grails.serverURL` or keep it empty, otherwise redirect URL won't be calculated with context

**Comment by parthbhagat** [comment]
There is one more finding I came across, may be this is helpful to verify the fix.

I tried with grails 4.0.9, with this version created a main project and a grails plugin. With this setup when we run main project from command line then it works fine and <g:include> tag from the grails plugin shows the content from correct resource.

When the same setup is been deployed on wildfly (after removing tomcat dependency '`spring-boot-starter-tomcat`' - conflict with wildfly) then <g:include> tag does not work.
